app/src/main/java/ch/epfl/sdp/appart/glide/visitor/GlideLoaderVisitorImpl.java

```diff
@@ -0,0 +1,40 @@
+package ch.epfl.sdp.appart.glide.visitor;
+
+import android.content.Context;
+import android.widget.ImageView;
+
+import com.bumptech.glide.Glide;
+import com.google.firebase.storage.FirebaseStorage;
+import com.google.firebase.storage.StorageReference;
+
+import ch.epfl.sdp.appart.database.FirebaseDB;
+import ch.epfl.sdp.appart.database.MockDataBase;
+
+public final class GlideLoaderVisitorImpl implements GlideLoaderVisitor {
+
+    private final ImageView view;
+    private final String imageReference;
+    private final Context context;
+
+    public GlideLoaderVisitorImpl(Context context, ImageView view, String imageReference) {
+        this.context = context;
+        this.view = view;
+        this.imageReference = imageReference;
+    }
+
+    @Override
+    public void visit(FirebaseDB database) {
+        StorageReference ref = FirebaseStorage.getInstance().getReferenceFromUrl("gs://appart-ec344.appspot.com/Cards/" + imageReference);
```

Comment:
can't we use `FirebaseStorage.getInstance()` instead of the hardcoded url? Not a major issue but it would be cleaner

Reply:
yes, thank you I will update it.